### PR TITLE
Rework 'Contributing' section

### PR DESCRIPTION
**Pull request by martinvanhensbergen**
The 'Contributing' section is separated in three parts, discussing process, style guide and tooling.

'Developing for Dissect' now describes the contributing process in more detail, including the Contributor License Agreement and principle of least dependencies.

'Style guide' has been reworded for clarity and provided with extra headings to better discern the different topics.

'Tooling' now includes information about tooling that was under a different section previously and also adds information in the upcoming tooling to test documentation.

### CODE REVIEW

File: docs/source/contributing/developing.rst [hunk 1]
```diff
@@ -5,87 +5,125 @@ Developing for Dissect
...
+
+
+Development process
+-------------------
+
+
+Python and tox versions
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Dissect is built and tested against Python 3.9 (CPython and PyPy). Older versions may not work, as features are used which
```
Comment (Schamper) [inline docs/source/contributing/developing.rst:30]:
```suggestion
Dissect is built and tested against Python 3.9, 3.10 and 3.11 (CPython and PyPy). Older versions may not work, as features are used which
```
We also test against 3.10 and 3.11 these days.

Comment (pyrco) [reply docs/source/contributing/developing.rst:30]: But not PyPy, that is only 3.9.

Comment (Schamper) [reply docs/source/contributing/developing.rst:30]: Updated the suggestion a bit to reflect that, but feel free to just completely rewrite it however you see fit @martinvanhensbergen :)

**Review by pyrco — CHANGES_REQUESTED** [review]

File: docs/source/contributing/developing.rst (hunk 1 above)
Comment (martinvanhensbergen) [reply docs/source/contributing/developing.rst]: I have changed this section somewhat; please check again.

File: docs/source/contributing/developing.rst [hunk 2]
```diff
@@ -5,87 +5,125 @@ Developing for Dissect
...
 may not yet be supported by these versions (for example ``@cached_property`` is only supported since Python 3.9).
 Newer versions will probably work, but are not guaranteed to.
 
 To build and test Dissect projects, `tox <https://tox.wiki/en/latest/>`_ is used. A minimum version of 3.8 is required.
```
[the remark this answers is not in the document]
Comment (martinvanhensbergen) [reply docs/source/contributing/developing.rst]: I have incorporated this information as well.

File: docs/source/contributing/developing.rst (hunk 1 above)
Comment (Schamper) [reply docs/source/contributing/developing.rst]: LGTM

**Review by pyrco — APPROVED** [review]

**Review by pyrco — APPROVED** [review]

**Pull request merged**